README.md

```diff
@@ -1,130 +1,209 @@
 
-1. User clicks "Connect your car" button (or similar) on your application's website.
-2. The user is redirected to a new page, either as a popup (if using the `openDialog` and/or `addClickHandler` methods), or in the same page (if the client redirects to the link generated by `generateLink` directly, without using `openDialog` and/or `addClickHandler`). This page requires the user authenticate with their vehicle credentials.
-3. The user will be asked to authorize your application to connect to their vehicle.
-4. Smartcar's services will redirect back to your application's redirectUri. The redirect will include an authorization code if the user approved the authorization, an error otherwise. An optional callback function (passed in as `onComplete`) will be invoked if `callback.js` is imported in the page served at your application's redirectUri.
-5. Your application's backend server will need to accept the authorization code and exchange it for an access token.
+    *Note:* Due to the implementation of `redirect.js`, the redirect page must be opened in a separate window.
+3. After entering their credentials, the user will be presented with a list of their vehicles to authorize and a selection of permissions to grant your app.
+4. Once the user has selected their vehicle and selected permissions, this SDK will automagically trigger the `onComplete` function you provide with the authorization code or an error. If called with a code, you must send this code to your back end to then exchange it with Smartcar for an access token.
 
-The SDK will help facilitate the OAuth link generation, popup dialog creation, and Smartcar will handle the user authentication and authorization. This SDK will not assist with the backend server code to accept authorization codes or exchanging for access tokens (step 6).
+    *Note:* Okay, so, about that automagically... Here's what happens behind the scenes: Let's say your website is `https://my-amazing-app.com`. Following the OAuth flow the user is redirected to a Smartcar-hosted redirect URI `https://javascript-sdk.smartcar.com/redirect-2.0.0?app_origin=https://my-amazing-app.com&code={authorization-code}`. This page sources `redirect.js` which uses `postMessage` to send a message with the authorization code or error to the origin specified by the `app_origin` query param `https://my-amazing-app.com`. The redirect page then closes itself out. On `https://my-amazing-app.com` the Smartcar instance you created receives the posted message and fires your `onComplete` method with the appropriate arguments. From your user's perspective, all they'll see is a brief flicker of the redirect page before it closes out, depending on their connection speed.
+5. Your application's back end server will need to accept the authorization code and exchange it for an access token.
```

Comment:
You will need to send the authorization code to your back and have it exchange the code for an access token.